src/frontend/components/VideoPlayer/index.spec.tsx

```diff
@@ -0,0 +1,369 @@
+      },
+      mp4: {
+        144: 'https://example.com/144p.mp4',
+        1080: 'https://example.com/1080p.mp4',
+      },
+      thumbnails: {
+        720: 'https://example.com/144p.jpg',
+      },
+    },
+  } as Video;
+
+  const createPlayer = jest.fn(() => ({
+    destroy: jest.fn(),
+  }));
+
+  it('starts up the player with DashJS and renders all the relevant sources', () => {
+    // Simulate a browser that supports MSE and will use DashJS
+    mockIsMSESupported.mockReturnValue(true);
+    const state = {
+      jwt: 'jwt-token',
+      state: appState.INSTRUCTOR,
+      video,
+    } as any;
+
+    const { container, getByText, queryByText } = render(
+      <Provider store={bootstrapStore(state)}>
+        <VideoPlayer createPlayer={createPlayer} video={video} />
+      </Provider>,
+    );
+
+    // The player is created and initialized with DashJS for adaptive bitrate
+    expect(createPlayer).toHaveBeenCalledWith(
+      'plyr',
+      expect.any(Element),
+      'jwt-token',
+      expect.anything(),
+    );
+    expect(mockInitialize).toHaveBeenCalledWith(
+      expect.any(Element),
+      'https://example.com/dash.mpd',
+      false,
+    );
+    expect(mockSetInitialBitrateFor).toHaveBeenCalledWith('video', 1600000);
+    expect(queryByText(/Download this video/i)).toEqual(null);
+    getByText('Show a transcript');
+    expect(container.querySelectorAll('track')).toHaveLength(2);
+  });
+
+  it('allows video download when the video object specifies it', () => {
+    mockIsMSESupported.mockReturnValue(false);
+    const state = {
+      jwt: 'jwt-token',
+      state: appState.INSTRUCTOR,
+      video: {
+        ...video,
+        show_download: true,
+      },
+    } as any;
+
+    const { container, getByText, queryByText } = render(
+      <Provider store={bootstrapStore(state)}>
+        <VideoPlayer createPlayer={createPlayer} video={video} />
+      </Provider>,
+    );
+
+    getByText(/Download this video/i);
+    getByText('Show a transcript');
+  });
+
+  it('does not use DashJS when MSE are not supported', () => {
+    // Simulate a browser that does not support MSE
+    mockIsMSESupported.mockReturnValue(false);
+    const state = {
+      jwt: 'jwt-token',
+      state: appState.INSTRUCTOR,
+      video,
+    } as any;
+
+    const { container } = render(
+      <Provider store={bootstrapStore(state)}>
+        <VideoPlayer createPlayer={createPlayer} video={video} />
+      </Provider>,
+    );
+
+    // The player is created and initialized with DashJS for adaptive bitrate
+    expect(createPlayer).toHaveBeenCalledWith(
+      'plyr',
+      expect.any(Element),
+      'jwt-token',
+      expect.anything(),
+    );
+    expect(mockInitialize).not.toHaveBeenCalled();
+    expect(mockSetInitialBitrateFor).not.toHaveBeenCalled();
+    expect(container.querySelectorAll('source[type="video/mp4"]')).toHaveLength(
+      2,
+    );
+  });
+
+  describe('mapStateToProps()', () => {
```

Comment:
IMO one of the advantages of bundling connectivity with the component is to stop testing irrelevant internals like these.

🔥🔥🔥